### ISSUE TITLE
Treat items starting with '_' the same as files starting with '.' and ignore them.

### ISSUE THREAD

**Issue by Terraphice** [Peltoche/lsd#751]
A relatively uncommon but still popular method of marking files to be ignored is beginning them with '*_*'  instead of '.'.

VIM does this, with the *_*viminfo file, less does this for the *_*lesshist file, etc. It's pretty uncommon, but it's also really uncommon to begin files/directories with '*_*' in the first place.

I'd like to see this prefix added alongside '.' and ignored like it is in exa. It's the only thing exa does differently that I really want in lsd.

Example:
![image](https://user-images.githubusercontent.com/41020526/195320017-fcd3f18f-8d42-4d25-8679-3c4f9711f1d8.png)

**Comment by meain** [comment]
I don't think that is something we would want in `lsd` as this would make us deviate from what gnu ls does. Do you have some links pointing to docs in exa where they mention hiding filenames starting with `_`?

**Comment by Terraphice** [comment]
I don't really feel like digging through docs. Here's the difference in action though: 
![image](https://user-images.githubusercontent.com/41020526/195330633-09f6f25d-7dff-45ce-8a58-c732aa4ef5fa.png)

**Comment by zwpaper** [comment]
maybe we can did this for windows, seems like a windows feature/common used case

https://stackoverflow.com/a/63516223/4242336

**Comment by Terraphice** [comment]
Any updates to this issue that I might have missed? It's been quite a while and this issue was never revisited.

**Comment by zwpaper** [comment]
we could hide `_` prefixed file ONLY in windows by default, and add a flag to control this.

ref: https://man7.org/linux/man-pages/man1/ls.1.html

```
       --hide=PATTERN
              do not list implied entries matching shell PATTERN
              (overridden by -a or -A)
```

**Comment by ChrisDenton** [comment]
As a Windows user, I'd be very surprised if this behaviour were enabled by default. It would seem very weird to me if, e.g. `__init__.py` files were being hidden. I get the vim use case and I understand making it configurable but I feel it should be something users choose to opt-in to as it's a niche non-standard thing that most people won't even be aware of.

**Comment by ofersadan85** [comment]
This seems like a really bad idea to implement automatically, I don't know any other program that does this on Windows, other than VIM which isn't exactly a Windows native program. Preferably just leave it to the users to use "ignore-globs"

**Comment by Terraphice** [comment]
Yeah I don't particularly care whether it's automatic or not, I just think the feature should be there in some capacity, especially if you want feature parity with exa/eza or whatever.

**Comment by ofersadan85** [comment]
So in other words, you suggest using "ignore-globs" for ignoring paths, but when doing `-a` (which implies show everything) the ignores should not be used. Is that right? @Terraphice ? I think this makes sense, especially since we already have the middle ground of using `-A` which shows "almost everything" so that *should* ignore what the user wanted to ignore, but `-a` wouldn't. 

If @zwpaper / @meain support this, I'll happily write a PR for this

**Comment by zwpaper** [comment]
yes, IMHO just like the `GNU ls` support `--ignore` and `--hide`, we should also impl that and leave the choice to users, they could add any globs to the configuration file if they like.

and we are not setting any default values even in Windows.

please notice, 

- `--ignore` will NOT be overwritten by `-a` or `-A`
- `--hide` will be overwritten by both `-a` and `-A`

thanks so much if anyone could help @ofersadan85

**Comment by Terraphice** [comment]
It's been 2 years since I opened the issue and 9 months since the last reply, so I'll bump this.
